autotst/job/job.py

```diff
@@ -323,108 +376,220 @@ def check_isomorphic(conformer):
             else:
                 return check_isomorphic(conformer)
 
-    def calculate_species(self, species):
+        raise Exception("Shoudn't reach here")
+
+    def calculate_fod(self,conformer):
+        """
+        Runs finite temperaure DFT to determine the Fractional Occupation number weighted Density (FOD number).
+        Uses Orca's the default functional, basis set, and smear temperature (TPSS, def2-TZVP, 5000 K).
+        """
+        # specify a directory for FOD input and output
+        if isinstance(conformer,TS):
+            fod_dir = os.path.join(self.directory, "ts",
+                                   conformer.reaction_label, "fod")
+            dir = os.path.join(self.directory, "ts",
+                               conformer.reaction_label)
+            label = conformer.reaction_label + "_fod"
+        else:
+            fod_dir = os.path.join(self.directory, "species", conformer.smiles, "fod")
+            dir = os.path.join(self.directory, "species",
+                               conformer.smiles)
+            label = conformer.smiles + "_fod"
+
+        if not os.path.exists(fod_dir):
+            os.makedirs(fod_dir)
+
+        # Get orca calculator instance 
+        # for lowest energy conformer
+        orca_calc = Orca(conformer=conformer)
+        orca_calc.write_fod_input(fod_dir)
+
+        # Assign FOD label for calulation and filepath 
+        # to save input and output
+        file_path = os.path.join(fod_dir, label)
+
+        # Assign environment variables with orca command and path
+        os.environ["COMMAND"] = orca_calc.command
+        os.environ["FILE_PATH"] = file_path
+
+        # Do not run orca if log file is already there
+        attempted = False
+        complete = False
+        if os.path.exists(file_path + ".log"):
+            attempted = True
+            logging.info(
+                "It appears that this job has already been run")
+            if orca_calc.check_NormalTermination(file_path + ".log"):
+                complete = True
+                logging.info("The FOD calculation completed! The FOD log file is {}".format(
+                    file_path + ".log"))
+                copyfile(
+                    file_path + ".log", os.path.join(dir, label + ".log"))
+                return True
+            else:
+                logging.info("It appears the FOD job did not terminate normally! Trying FOD job again")
+                complete = False
+
+        # In log file does not exist, run Orca
+        if not attempted or not complete:
+            logging.info(
+                "Starting FOD calculation for {}".format(conformer))
+            subprocess.call(
+                """sbatch --exclude=c5003,c3040 --job-name="{0}" --output="{0}.log" --error="{0}.slurm.log" -p test -N 1 -n 4 -t 01:00:00 --mem=1GB $AUTOTST/autotst/job/orca_submit.sh""".format(
```

Comment:
I'm not a fan of the `orca_submit.sh` file. It seems like we can make this cleaner.